packages/flutter/lib/src/widgets/app.dart

```diff
@@ -149,6 +166,48 @@ class WidgetsApp extends StatefulWidget {
   /// for this application's [Localizations] widget.
   final Iterable<LocalizationsDelegate<dynamic>> localizationsDelegates;
 
+  /// This callback is responsible for choosing the app's locale
+  /// when the app is started, and when the user changes the
+  /// device's locale.
+  ///
+  /// The returned value becomes the locale of this app's [Localizations]
+  /// widget. The callback's `locale` parameter is the device's locale when
+  /// the app started, or the device locale the user selected after the app was
+  /// started. The callback's `supportedLocales` parameter is just the value
+  /// [supportedLocales].
+  ///
+  /// If the callback is null then the resolved locale is:
+  /// - The callback's `locale` parameter if it's equal to a supported locale.
+  /// - The first supported locale with the same [Locale.langaugeCode] as the
+  ///   callback's `locale` parameter.
+  /// - The first locale in [supportedLocales].
+  ///
+  /// See also:
+  ///
+  ///  * [MaterialApp.resolveLocaleCallback], which sets the callback of the
+  ///    [WidgetsApp] it creates.
+  final ResolveLocaleCallback resolveLocaleCallback;
+
+  /// The list of locales that this app has been localized for.
+  ///
+  /// By default only the American English locale is supported. Apps should
+  /// configure this list to match the locales they support.
+  ///
+  /// This list must not null. It's default value is just
+  /// `[const Locale('en', 'US')]`.
+  ///
```

Comment:
this should mention that the order matters, in that the first will be the ultimate fallback, and that within the list if there's locales with country codes, the first locale with a given language will be used for all countries if no exact match could be found. e.g. if the list is 'en-GB', 'en-CA', 'en', and the user's language is 'en-US', we'll pick 'en-GB', not 'en'.